react/components/Block/index.js

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import Types from 'react/components/Block/util/Types';
+import Channel from 'react/components/Block/components/Types/Channel';
+
+const BlockWrapper = styled.div`
+  background: #fff;
+  display: inline-block;
+  flex: 1 1 285px;
+  max-width: 315px;
+  position: relative;
+  text-align: center;
+  user-select: none;
+  width: 100%;
+`;
+
+const BlockPadding = styled.div`
+  padding-top: 100%;
+`;
+
+const Block = (props) => {
+  const {
+    blockData,
+    type,
+  } = props;
+
+  const blockContent = () => {
+    let BlockTypeKlass;
+
+    switch (type) {
+      case Types.CHANNEL:
+        BlockTypeKlass = Channel;
+        break;
+      // TODO: Fill in the rest...
+      default:
+        console.error(`Invalid Block Type ${type} for Block Component.`);
+        return null;
+    }
+
+    return (
+      <BlockTypeKlass
+        {...blockData}
+      />
+    );
+  };
+
+  return (
+    <BlockWrapper>
+      <BlockPadding />
+      {blockContent()}
+    </BlockWrapper>
+  );
+};
+
+Block.propTypes = {
+  blockData: PropTypes.shape({}).isRequired,
```

Comment:
This should be validated with an actual data shape using the the proptype of the fragment used to fulfill it. 

a la
https://github.com/aredotna/ervell/blob/497d8f3bfbb1e6895dfa5745333f9359b1b0a9e0/react/components/ChannelMetadata/components/ChannelMetadataActions/index.js#L22

Reply:
Oh, I see this is the like top-level block—I suppose it could be something like a oneOf(shape) sort of thing.

Reply:
Yeah this is more "any props you want to send to the thing being plugged into the block (e.g.; channel, video, image, URL, etc). It's kind of just a catch-all object. Not much to validate on it and should be done on the block type itself.